lib/https.js

```diff
@@ -58,7 +58,22 @@ function createConnection(port, host, options) {
   }
 
   debug('createConnection', options);
-  return tls.connect(options);
+
+  var self = this;
+  if (options._agentKey && this._tlsSessions[options._agentKey]) {
+    debug('reuse session for %j', options._agentKey);
+    options = util._extend({
+      session: this._tlsSessions[options._agentKey]
```

Comment:
If `session` is the only one being set, maybe it is worth it to just use `options.session = options.session || ...`?